docs/interop.md

```diff
@@ -0,0 +1,151 @@
+
+#### Strings
+The [MarshalAs](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.marshalasattribute?view=netcore-3.1) attribute no longer supports [UnmanagedType.HString](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.unmanagedtype?view=netcore-3.1).  Instead, strings should be marshaled with the C#/WinRT MarshalString class.
+```csharp
+// public void SetString([MarshalAs(UnmanagedType.HString)] String s);
+public void SetString(IntPtr hstr);
+// public void GetString([MarshalAs(UnmanagedType.HString)] out String s);
+public void GetString(IntPtr out hstr);
+// ...
+// Marshal HSTRING to System.String
+IntPtr hstr;
+GetString(out hstr);
+var str = MarshalString.FromAbi(hstr);
+// Marshal System.String as fast-pass HSTRING reference
+var marshalStr = MarshalString.CreateMarshaler("String");
+SetString(MarshalString.GetAbi(marshalStr));
+```
+
+#### IInspectables
+The runtime no longer supports marshaling IInspectable-based ComImport interfaces. But these can be approximated with IUnknown:
+```csharp
+[ComImport]
+[InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+[Guid("68B3A2DF-8173-539F-B524-C8A2348F5AFB")]
+internal unsafe interface IServiceProviderInterop
+{
+    // Note: Invoking methods on ComInterfaceType.InterfaceIsIInspectable interfaces
+    // is no longer supported in the runtime, but can be simulated with IUnknown.
+    void GetIids(out int iidCount, out IntPtr iids);
+    void GetRuntimeClassName(out IntPtr className);
+    void GetTrustLevel(out TrustLevel trustLevel);
+
+    void GetService(IntPtr type, out IntPtr service);
+}
+```
+
+
+## Create RCW
+The Marshal RCW creation functions:
+```csharp
+T obj = (T)Marshal.GetObjectForIUnknown(ptr);
+T obj = Marshal.GetTypedObjectForIUnknown<T>(ptr);
+```
+can be replaced with the appropriate projected type's (or marshaler's) FromAbi method:
+```csharp
+T obj = ABI.Namespace.T.FromAbi(ptr);
+T obj = MarshalInterface<T>.FromAbi(ptr);
+T obj = MarshalInspectable<T>.FromAbi(ptr);
+T obj = MarshalGeneric<T>.FromAbi(ptr);
+```
+
+## Create CCW
+The Marshal CCW creation function:
+```csharp
+Marshal.GetComInterfaceForObject(obj)
+```
+can be replaced with the appropriate projected type's (or marshaler's) FromManaged method, which supports ICustomQueryInterface.
+```csharp
+IntPtr ptr = ABI.Namespace.T.FromManaged(obj);
+IntPtr ptr = MarshalInterface<T>.FromManaged(obj);
+IntPtr ptr = MarshalInspectable<T>.FromManaged(obj);
+IntPtr ptr = MarshalGeneric<T>.FromManaged(obj);
+```
+
+## Unsupported
+The following Marshal CCW/RCW-related functions are not supported in C#/WinRT:
+```csharp
+AreComObjectsAvailableForCleanup,
+ChangeWrapperHandleStrength
+CleanupUnusedObjectsInCurrentContext
+CreateAggregatedObject
+CreateWrapperOfType
+GetComObjectData
+SetComObjectData

```

Comment:
The IWinRTObject.AdditionalTypeData dictionary could be considered to replace these two members.

Reply:
oh sure - good suggestion